### PR TITLE
Add lang and version labels to containers

### PR DESCRIPTION
**Pull request by eddumelendez**
The following labels are added `org.testcontainers.lang` and
`org.testcontainers.version`. Those are good identifiers when debugging
and sharing logs.

### CODE REVIEW

**Review by kiview — DISMISSED** [review]

**Review by mdelapenya — APPROVED** [review]

File: core/src/main/java/org/testcontainers/utility/ResourceReaper.java [hunk 1]
```diff
@@ -41,10 +42,18 @@ public class ResourceReaper {
...
+    static Map<String, String> markerLabels() {
+        Map<String, String> labels = new HashMap<>();
+        labels.put(DockerClientFactory.TESTCONTAINERS_SESSION_ID_LABEL, DockerClientFactory.SESSION_ID);
+        labels.put(DockerClientFactory.TESTCONTAINERS_LANG_LABEL, "java");
+        labels.put(
+            DockerClientFactory.TESTCONTAINERS_VERSION_LABEL,
+            ResourceReaper.class.getPackage().getImplementationVersion()
+        );
+        return Collections.unmodifiableMap(labels);
+    }
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/utility/ResourceReaper.java:56]: This place isn't right for adding these labels (e.g. reusable containers won't get marked). We should add them here instead:

https://github.com/testcontainers/testcontainers-java/blob/29aa84d19739f3d20c485275180e0619dd4145b1/core/src/main/java/org/testcontainers/DockerClientFactory.java#L61

**Review by bsideup — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/testcontainers/DockerClientFactory.java [hunk 2]
```diff
@@ -56,9 +56,24 @@ public class DockerClientFactory {
...
+
+    public static final String TESTCONTAINERS_VERSION_LABEL = TESTCONTAINERS_LABEL + ".version";
+
     public static final String SESSION_ID = UUID.randomUUID().toString();
 
-    public static final Map<String, String> DEFAULT_LABELS = ImmutableMap.of(TESTCONTAINERS_LABEL, "true");
+    public static final Map<String, String> DEFAULT_LABELS = markerLabels();
+
+    static Map<String, String> markerLabels() {
+        String implementationVersion = ResourceReaper.class.getPackage().getImplementationVersion();
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/DockerClientFactory.java:68]:
```suggestion
        String implementationVersion = DockerClientFactory.class.getPackage().getImplementationVersion();
```

File: core/src/main/java/org/testcontainers/DockerClientFactory.java [hunk 3]
```diff
@@ -56,9 +56,24 @@ public class DockerClientFactory {
...
+        String testcontainersVersion = implementationVersion == null ? "unspecified" : implementationVersion;
+
+        Map<String, String> labels = new HashMap<>();
+        labels.put(TESTCONTAINERS_LABEL, "true");
+        labels.put(DockerClientFactory.TESTCONTAINERS_LANG_LABEL, "java");
+        labels.put(DockerClientFactory.TESTCONTAINERS_VERSION_LABEL, testcontainersVersion);
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/DockerClientFactory.java:74]:
```suggestion
        labels.put(TESTCONTAINERS_LANG_LABEL, "java");
        labels.put(TESTCONTAINERS_VERSION_LABEL, testcontainersVersion);
```

**Review by bsideup — APPROVED** [review]
approving with a couple of nits 👍

**Comment by eddumelendez** [comment]
good catch @bsideup! thanks!

**Pull request merged**